Added the metrics plugin to the RPM/DEB packages

rogaha opened the pull request:
**Packages:**

 - centos
 - fedora
 - ubuntu
 - debian

**the plugin is disabled by default**

Signed-off-by: Roberto Gandolfo Hashioka <roberto_hashioka@hotmail.com>

FrenchBen commented on deb/common/docker-ce.docker.upstart:
```diff
@@ -69,4 +83,47 @@ post-start script
 	fi
+
+	# store the metrics plugin state
+		plugin_name=$(docker plugin ls | grep docker/metrics-plugin | awk 'NR==1{print $2}')
```
fix indentation

FrenchBen commented on rpm/centos-7/docker-ce.spec:
```diff
@@ -6,6 +6,7 @@ Group: Tools/Docker
 License: ASL 2.0
 Source0: engine.tgz
 Source1: cli.tgz
+Source2: metrics-plugin_%{_version}-stable-%{_plugin_version}.tgz
```
can `-stable` be part of the `_plugin_version` so that an `edge` version of the plugin could be installed?

rogaha replied:
👍

rogaha commented on deb/Makefile:
```diff
@@ -37,54 +46,66 @@ ubuntu-trusty: ## build ubuntu trusty deb packages
 debian-stretch: ## build debian stretch deb packages
 	docker build -t debbuild-$@/$(ARCH) -f $(CURDIR)/$@/Dockerfile.$(ARCH) .
 	docker run --rm \
+	  -e PLUGIN_VERSION=$(PLUGIN_VERSION) \
```
:+1

rogaha commented:
thanks for the feedback @FrenchBen. I've addressed all your comments.

andrewhsu commented on rpm/centos-7/Dockerfile.x86_64:
```diff
@@ -18,6 +18,7 @@ RUN yum install -y \
    vim-common
 
 ENV GO_VERSION 1.7.5
+ENV DISTRO centos
```
can you split the `DISTRO` line changes in the dockerfiles into a separate PR so they can be merged in first?

rogaha replied:
👍 I'll do that.

rogaha replied:
PR submitted @andrewhsu: https://github.com/docker/docker-ce-packaging/pull/5

andrewhsu commented:
As discussed with @rogaha the systemd service files will be pulled into this repo since it is specific to the linux distros that docker-ce will be installed on. I'll create a PR for this.

rogaha commented:
thanks for the update @andrewhsu. I'll update the PR accordingly as soon as the `systemd` scripts are added the the `repo`.

rogaha commented:
@andrewhsu I've moved the `DISTRO/SUITE` changes to https://github.com/docker/docker-ce-packaging/pull/5.

rogaha commented:
ping @andrewhsu

rogaha commented:
I've tested it the following commands:

### DEB
```
make DOCKER_BUILD_PKGS='ubuntu-xenial' VERSION=17.06.0 PLUGIN_VERSION=1.0.0-test deb
```

### RPM
```
make DOCKER_BUILD_PKGS='centos-7' VERSION=17.06.0-dev PLUGIN_VERSION=1.0.0-test rpm
```

rogaha commented:
@andrewhsu PTAL

andrewhsu commented:
Still need to address how the plugin will be added to the rpm and deb packages for non-x86_64 architectures.

rogaha commented:
@andrewhsu I've added support to `armv7l, s390x and x86_64` to the `metrics plugin` (https://github.com/docker/metrics-plugin/pull/17) and I've updated this PR accordingly, PTAL.

rogaha commented:
@andrewhsu PR rebased.

andrewhsu commented:
LGTM

Built centos-7 pkg:
```
$ make ENGINE_DIR=/home/ubuntu/src/docker-ce/components/engine CLI_DIR=/home/ubuntu/src/docker-ce/components/cli centos-7
$ find rpmbuild/ -type f -name '*.rpm' | grep RPM
rpmbuild/SRPMS/docker-ce-17.07.0-0.0.20170720.233508.git00313efde4.el7.centos.src.rpm
rpmbuild/RPMS/x86_64/docker-ce-debuginfo-17.07.0-0.0.20170720.233508.git00313efde4.el7.centos.x86_64.rpm
rpmbuild/RPMS/x86_64/docker-ce-17.07.0-0.0.20170720.233508.git00313efde4.el7.centos.x86_64.rpm
```

seemethere commented:
@rogaha Getting an error right now when trying to build rpm packages:

```~/work/docker-ce-packaging/rpm master ⇡
❯ make ENGINE_DIR="$HOME/work/docker-ce/components/engine" CLI_DIR="$HOME/work/docker-ce/components/cli" fedora-25
mkdir -p rpmbuild/SOURCES
mkdir -p /Users/eliasuriegas/work/docker-ce-packaging/rpm/plugin
docker create --name tmp docker/telemetry-tgz:1.0.0.linux-x86_64-test
Unable to find image 'docker/telemetry-tgz:1.0.0.linux-x86_64-test' locally
Error response from daemon: repository docker/telemetry-tgz not found: does not exist or no pull access
make: *** [rpmbuild/SOURCES/telemetry.tgz] Error 1```

rogaha commented:
@seemethere can you please confirm the `username` you are using to `pull` the images?

rogaha commented:
it should be working now.
